kernel/aster-nix/src/process/posix_thread/futex.rs

```diff
@@ -423,17 +431,31 @@ impl FutexWaiter {
         }
     }
 
-    pub fn wait(&self) {
+    pub fn wait(&self, timeout: Option<FutexTimeout>) {
         let current_thread = current_thread!();
         if current_thread.tid() != self.tid {
             return;
         }
         self.is_woken.store(false, Ordering::SeqCst);
-        while !self.is_woken() {
-            // debug!("futex is wait for waken, tid = {}", self.tid);
-            Thread::yield_now();
+
+        if let Some(timeout) = timeout {
+            let wait_queue = WaitQueue::new();
+            let wake_cond = || {
+                if self.is_woken() {
+                    Some(())
+                } else {
+                    None
+                }
+            };
+
+            wait_queue.wait_until_or_timeout(wake_cond, &timeout.duration);
+        } else {
+            // The PtyTest.TermiosPollReplica test will stuck if we use WaitQueue.
+            while !self.is_woken() {
```

Comment:
Could you please try again whether the branch can be removed when doing modifications according to the former comment?